python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -220,11 +239,37 @@ def _create_repository_using_definitions_args(
 
     check.opt_mapping_param(loggers, "loggers", key_type=str, value_type=LoggerDefinition)
 
-    (
-        jobs_with_resources,
-        schedules_with_resources,
-        sensors_with_resources,
-    ) = _attach_resources_to_jobs_and_instigators(jobs, schedules, sensors, resource_defs)
+    if isinstance(jobs, BindResourcesToJobs):
+        (
+            jobs_with_resources,
+            schedules_with_resources,
+            sensors_with_resources,
+        ) = _attach_resources_to_jobs_and_instigators(jobs, schedules, sensors, resource_defs)
```

Comment:
so right now if `BindResourcesToJobs` wraps the jobs then it will also attach the resources to sensors and schedules. So in effect `BindResourcesToJobs` acts as a definitions-wide global even though it appears to be scope to jobs only. I think I am reading this right.

Reply:
Maybe the helper method here could be clarified - what this does is updates all the jobs and the instigators' jobs (if the instigators have `JobDefinitions` directly attached, they also need to be updated). This is separate from binding resources to a schedule or sensor function.